### DISCUSSION TITLE
Using sea-orm entities with poem-openapi

### DISCUSSION THREAD

**Discussion by zchira** [poem-web/poem#710] [Q&A] [answered]
Simple question:
Let's say we have database with few tables. Let's use generic example for tables: Author and Posts (one-to-many relation).
With `sea-orm-cli` we can generate entities structs `AuthorEntity` and `PostEntity`.

On the other side, I have poem-openapi endpoints:
GET `/authors`  - return all authors 
GET `/authors/:authorId/posts` - return all posts from author identified with `authorId`

What would be the cleanest way to do that?
I guess, from the endpoint I can't directly return structs `AuthorEntity` and `PostEntity` since they don't have `#derive(Object)`?
The other way would be to create my own structs:
```

#derive(Object) 
struct Author {
...
}

#derive(Object) 
struct Post {
...
}

```

and to implement From trait for AuthorEntity and PostEntity for them:
```

impl From<AuthorEntity> for Author {
...
}

impl From<PostEntity> for Post {
...
}


```

Is there some more elegant/clean/rusty way?

**Comment by chikko80** [comment]
I am not familiar with `sea-orm` but what about just adding the `Object` derive to the `AuthorEntity`?

You can even do this on generation with `--model-extra-derives`

**Comment by zchira** [comment]
Thanks @chikko80 ,
that makes sense.

I tried to add `--model-extra-derives Object` and the generated entity has:
`#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Object)]`

But, since there is no `use poem_openapi_derive::Object` compile error is thrown.

I tried also with:
 `--model-extra-derives poem_openapi_derive::Object`
but the generated entity looks bad:
`#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, poem_openapi :: Object)]` (<< spaces before and afte `::`).

I'll check the docs for sea-orm-cli to see if there are some other options :+1:

**Reply by chikko80** [reply, marked as answer]
Otherwise reach out to the seaorm devs / open a question/issue.
I've heard the team is quit responsive and always happy to get suggestions for improvements

**Reply by chikko80** [reply]
@zchira 

would you mind pinging me here if you have a solution for this?
I am planning to try out seaorm in an upcoming project and would face the same issue

**Reply by zchira** [reply]
Sure, I asked on sea-orm discussions also ( [link](https://github.com/SeaQL/sea-orm/discussions/2033) )
